src/stores/Store.ts

```diff
@@ -74,6 +81,246 @@ export default class Store {
+			);
+		} else if (this.selectedSchemaType === 'act') {
+			this.setEditorProperties(
+				this.actMessagesHistory[index].actBox,
+				this.actMessagesHistory[index].fullServiceName,
+				this.actMessagesHistory[index].methodName,
+				this.actMessagesHistory[index].message as string,
+			);
+		}
+
+		this.setEditMessageMode(true);
+		this.setEditedMessageIndex(index);
+	};
+
+	setEditorProperties = (
+		sessionOrActBox: string | null,
+		dictionaryOrService: string | null,
+		messageTypeOrMethod: string | null,
+		editorCode: string,
+	) => {
+		if (this.selectedSchemaType === 'parsed-message') {
+			this.selectedSession = sessionOrActBox;
+			this.selectedDictionaryName = dictionaryOrService;
+			this.selectedMessageType = messageTypeOrMethod;
+		} else if (this.selectedSchemaType === 'act') {
+			this.selectedActBox = sessionOrActBox;
+			this.selectedService = dictionaryOrService;
+			this.setSelectedMethod(messageTypeOrMethod);
+		}
+
+		if (editorCode) {
+			this.setEditorCode(editorCode);
+		}
+	};
+
+	@action setEditorCode = (code: string) => {
+		this.editorCode = code;
+	};
+
+	@action getIndicatorByResult = (code: number): Indicator => {
+		if (code === 200) return 'indicator-successful';
+		return 'indicator-unsuccessful';
+	};
+
+	@action addIndicator = (indicatorClass: Indicator) => {
+		this.indicators.push(indicatorClass);
+	};
+
+	@action changeIndicator = (index: number, indicatorClass: Indicator) => {
+		this.indicators[index] = indicatorClass;
+	};
+
+	@action deleteIndicator = (index: number): Indicator[] => {
+		const tmpArray: Indicator[] = this.indicators.slice();
+		this.indicators = [];
+		tmpArray.forEach((item, i) => {
+			if (i !== index) {
+				this.addIndicator(item);
+			}
+		});
+		return this.indicators;
+	};
+
+	@computed get getCurrentMessagesArray() {
+		return this.selectedSchemaType === 'parsed-message'
+			? this.parsedMessagesHistory
+			: this.actMessagesHistory;
+	}
+
+	@action
+	public setPanelArea = (panelArea: number) => {
+		this.messageListPanelArea = panelArea;
+	};
+
+	@action setReplayMode = (flag: boolean) => {
+		this.isReplay = flag;
+	};
+
+	@action addParsedMessage = (
+		message: ParsedMessageItem | ActMessageItem,
+		indicatorClass?: Indicator,
+	) => {
+		this.addIndicator(indicatorClass || 'indicator-unvisible');
+		if (this.selectedSchemaType === 'parsed-message' && isParsedMessageItem(message)) {
+			this.parsedMessagesHistory.push(message);
+		} else if (this.selectedSchemaType === 'act' && isActMessageItem(message)) {
+			this.actMessagesHistory.push(message);
+		}
+	};
+
+	@action clearParsedMessages = () => {
+		if (this.selectedSchemaType === 'parsed-message') {
+			this.parsedMessagesHistory = [];
+		} else if (this.selectedSchemaType === 'act') {
+			this.actMessagesHistory = [];
+		}
+		this.indicators = [];
+
+		if (this.editMessageMode === true) {
+			this.setEditMessageMode(false);
+		}
+	};
+
+	@action startApp = () => {
```

Comment:
This code actually should be inside constuctor.
Also we can create LocalStorageWorker class to move all related to localStorage code there